workflow/workflow-demos-test/src_test/test/bpm/TestProcurementRequest.java

```diff
@@ -168,50 +162,34 @@ void approvalTask_verified_accepted(BpmClient bpmClient)
         .as().user("ldv")
         .execute();
     assertThat(result).isNotNull();
-    List<ITask> tasks = result.workflow().nextTasks();
-    assertThat(tasks).hasSize(2);
-    ITask verifyTask1 = tasks.get(0);
-    assertThat(verifyTask1.getName()).startsWith("Verify Request:");
-    assertThat(verifyTask1.getActivatorName()).isEqualTo("Teamleader");
-    ITask verifyTask2 = tasks.get(1);
-    assertThat(verifyTask2.getName()).startsWith("Verify Request:");
-    assertThat(verifyTask2.getActivatorName()).isEqualTo("Manager");
+    TaskSelector tasks = result.workflow().nextTask();
+    assertThat(result.workflow().nextTasks()).hasSize(2);
+    assertThat(tasks.activatorRole(TEAMLEADER).get().getName()).startsWith("Verify Request:");
+    assertThat(tasks.activatorRole(MANAGER).get().getName()).startsWith("Verify Request:");
     return tasks;
   }
-
-  private void verifyRequest(BpmClient bpmClient, ITask verifyTask, String role)
+  
+  private ExecutionResult verifyRequest(BpmClient bpmClient, TaskSelector tasks, String role)
   {
-    ExecutionResult result = bpmClient
-        .start().resumableTask(verifyTask)
-        .as().role(role)
-        .execute();
+    ExecutionResult result = bpmClient.start()
+            .resumableTask(tasks.activatorRole(role).get())
+            .as().role(role).execute();
     assertThat(result.workflow().task().getState()).isIn(TaskState.DONE, TaskState.READY_FOR_JOIN);
+    return result;
   }
 
-  private ProcurementRequest acceptRequest(BpmClient bpmClient, ITask acceptRequestTask)
+  private ProcurementRequest acceptRequest(BpmClient bpmClient, Workflow workflow)
```

Comment:
To be consistent I would provide here also a TaskSelector

Reply:
I think it would be better to accept here the last result, because other ways we need to call on every method call: result.workflow().nextTask().

```java
acceptRequest(bpmClient, result, MANAGER);
//instead of 
acceptRequest(bpmClient, result.workflow().nextTask(), MANAGER);
```

Reply:
yes